modules/reactor/src/meshgenerators/PatternedHexPeripheralModifier.C

```diff
@@ -0,0 +1,262 @@
+      "PatternedHexMeshGenerator and its derived classes such as "
+      "HexIDPatternedMeshGenerator.");
+  params.addRequiredParam<BoundaryName>("input_mesh_external_boundary",
+                                        "The external boundary of the input mesh.");
+  params.addRequiredParam<unsigned int>("new_num_sector",
+                                        "Number of sectors of each side for the new mesh.");
+  params.addParam<subdomain_id_type>(
+      "transition_layer_id",
+      TRANSITION_LAYER_DEFAULT,
+      "Optional customized block id for the transition layer block.");
+  params.addParam<SubdomainName>("transition_layer_name",
+                                 "transition_layer",
+                                 "Optional customized block name for the transition layer block.");
+  params.addRangeCheckedParam<unsigned int>(
+      "num_layers", 1, "num_layers>0", "Layers of elements for transition.");
+  params.addParam<std::vector<std::string>>("extra_id_names_to_modify",
+                                            "Names of the extra ids that need to be modified along "
+                                            "with the peripheral region modification.");
+  params.addParam<std::vector<dof_id_type>>(
+      "new_extra_id_values_to_assign",
+      "Optional extra ids to be assigned to the modified regions to override original values.");
+  params.addClassDescription("This PatternedHexPeripheralModifier object removes the outmost layer "
+                             "of the input mesh and add a "
+                             "transition layer mesh to facilitate stitching.");
+
+  return params;
+}
+
+PatternedHexPeripheralModifier::PatternedHexPeripheralModifier(const InputParameters & parameters)
+  : PolygonMeshGeneratorBase(parameters),
+    _input_name(getParam<MeshGeneratorName>("input")),
+    _input_mesh_external_boundary(getParam<BoundaryName>("input_mesh_external_boundary")),
+    _new_num_sector(getParam<unsigned int>("new_num_sector")),
+    _transition_layer_id(getParam<subdomain_id_type>("transition_layer_id")),
+    _transition_layer_name(getParam<SubdomainName>("transition_layer_name")),
+    _num_layers(getParam<unsigned int>("num_layers")),
+    _extra_id_names_to_modify(isParamValid("extra_id_names_to_modify")
+                                  ? getParam<std::vector<std::string>>("extra_id_names_to_modify")
+                                  : std::vector<std::string>()),
+    _new_extra_id_values_to_assign(
+        isParamValid("new_extra_id_values_to_assign")
+            ? getParam<std::vector<dof_id_type>>("new_extra_id_values_to_assign")
+            : std::vector<dof_id_type>()),
+    _pattern_pitch_meta(declareMeshProperty<Real>("pattern_pitch_meta", 0.0)),
+    // Use HexagonConcentricCircleAdaptiveBoundaryMeshGenerator for control drum meshing
+    _is_control_drum_meta(declareMeshProperty<bool>("is_control_drum_meta", false)),
```

Comment:
> so you're having to propagate this info for every modification of the mesh part?

Yes. The metadata needs to be propagated so that the mesh can be used by `PatternedHexMG`

Reply:
> and what if we re currently wrapping a control drum? Shouldnt we be propagating that metadata?
User will need to use `HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` for control drum. That MG adapters neighboring  meshes so that peripheral modification is not needed.